zulipterminal/config/themes.py

```diff
@@ -28,6 +28,79 @@
     'starred',
 }
 
+SOLAR_BASE = dict(
+    # Base03='#002b36',
+    # Base02='#073642',
+    # Base01='#586e75',
+    # Base00='#657b83',
+    # Base0='#839496',
+    # Base1='#93a1a1',
+    # Base2='#eee8d5',
+    # Base3='#fdf6e3',
+    # Yellow='#b58900',
+    # Orange='#cb4b16',
+    # Red='#dc322f',
+    # Magenta='#d33682',
+    # Violet='#6c71c4',
+    # Blue='#268bd2',
+    # Cyan='#2aa198',
+    # Green='#859900',
+
+    # Base03='#023',
+    # Base02='#034',
+    # Base01='#567',
+    # Base00='#678',
+    # Base0='#899',
+    # Base1='#9aa',
+    # Base2='#eed',
+    # Base3='#ffe',
+    # Yellow='#b80',
+    # Orange='#c41',
+    # Red='#d32',
+    # Magenta='#d38',
+    # Violet='#67c',
+    # Blue='#28d',
+    # Cyan='#2a9',
+    # Green='#890',
+# base03='#1c1c1c',
+# base02='#262626',
+# base01='#585858',
+# base00='#626262',
+# base0='#808080',
+# base1='#8a8a8a',
+# base2='#e4e4e4',
+# base3='#ffffd7',
+# yellow='#af8700',
+# orange='#d75f00',
+# red='#d70000',
+# magenta='#af005f',
+# violet=	'#5f5faf',
+# blue='#0087ff',
+# cyan='#00afaf',
+# green='#5f8700',
+
+Base03='#111',
```

Comment:
I tried `h234` and `h235`. The color is different from the `#111` and `#222` approximation.

Reply:
The only way to get `#1c1c1c` accurately is via `h234`. That is, assuming that h17-h255 is not tampered by the user in `.Xdefaults`. I also bumped the pending urwid 24-bit merge. The 24-bit version of solarized has chalkboard-like dark green bg color.

Reply:
`#262626`

Reply:
Testing full solarized base03: `#002b36`, base02: `#073642`.

Reply:
@rht We could use 'h' values, but according to the experience on #310 I think we should stick to hex for now, and we'll need a 256-color at least for the moment, and can add 24-bit later, but will still need the 256-color approximation?

If you agree with that:
* how is the use of colors?
* if you've seen solarized being used, are the colors good choices?
* do you see the expected named colors (as opposed to the grey in kitty (?) for selected, for example)

Reply:
Yet https://github.com/altercation/vim-colors-solarized/blob/master/colors/solarized.vim uses the `h###` version for the 256 mode. I think it is unlikely for h colors higher than 16 to be altered. Maybe put a warn message that says "if you alter your h1-h256, some of the colors in the theme will look different."

> how is the use of colors?

The background color is probably too black.

> do you see the expected named colors (as opposed to the grey in kitty (?) for selected, for example)

Selected messages are blue.

Reply:
@rht 
a) I think we should maintain equivalents in the code, which provide the best matches we can, for both cases, in case we want to provide a `h` mode but also a `#` mode for cases where terminal colors are varied? So we can list both in comments and default to one for the moment, perhaps.
b) There are two background colors? We could use the other (`Base02`) ?
c) Right; this is perhaps an orthogonal change, but with a lack of response on #310 I considered this an extra aspect to experiment with; do you like it?

Reply:
> I think we should maintain equivalents in the code, which provide the best matches we can, for both cases, in case we want to provide a `h mode` but also a `# mode` for cases where terminal colors are varied?

The `# mode` approximation is likely specific to Urwid (and might be stable across platforms). Though this means having to maintain 4 versions per theme (24-bit, `h mode`, `# mode`, 16 colors). Usually it is automatically selected to do 24-bit, fallback to h mode, then fallback to 16 colors. With 2 versions for 256-color, it's probably too many choices / too complicated for the users to have to check if the h mode is consistent with a screenshot then fallback to # mode.


> b) There are two background colors? We could use the other (Base02) ?

Base02 seems to be the main one. Base03 for stripes, e.g.
```
exe "hi! pandocTableZebraLight"              .s:fg_ptable.s:bg_base03.s:fmt_none
exe "hi! pandocTableZebraDark"               .s:fg_ptable.s:bg_base02.s:fmt_none
```

> c) Right; this is perhaps an orthogonal change, but with a lack of response on #310 I considered this an extra aspect to experiment with; do you like it?

#310 with # mode works on my terminal with green on msg_selected. But again, this is because previously the h mode used the number 1-16 for some of the colors.